feat: Added documentation for npm Enterprise installation on closed n…

mahikhond opened the pull request:
…etwork.

Details:
 - Created documentation for npm Enterprise installation by using VM's and Air-Gap solution.

mahikhond committed:
feat: Added documentation for npm Enterprise installation on closed network.

Details:
 - Created documentation for npm Enterprise installation by using VM's and Air-Gap solution.

arobson commented on tutorials/npm-Enterprise-on-a-closed-network.md:
```diff
@@ -0,0 +1,62 @@
+
+### Install npmE in open network and move VM’s to closed network
+
+To achieve this create two VM’s let’s say npmBlank and npmSource on open-network and install npm Enterprise on both these VM’s.
```
Please change to: 
To achieve this, create two VM's (`npmePrimary` and `npmeSource` for example) and install npm Enterprise on both using the instructions in the [Up and Running](https://npme.npmjs.com/docs/up-and-running/platforms/other.html) section.

In this configuration, publication of internal packages and all installs will happen against `npmePrimary` with `npmeSource` serving as the means for importing new public packages and new versions for existing public packages from the registry on a cadence you control.

arobson commented on tutorials/npm-Enterprise-on-a-closed-network.md:
```diff
@@ -0,0 +1,62 @@
+
+To achieve this create two VM’s let’s say npmBlank and npmSource on open-network and install npm Enterprise on both these VM’s.
+
+Both these VM’s originate as npm Enterprise servers, provisioned using the instructions provided [up and running section]( https://npme.npmjs.com/docs/up-and-running/platforms/other.html)
```
Remove this paragraph.

arobson commented on tutorials/npm-Enterprise-on-a-closed-network.md:
```diff
@@ -0,0 +1,62 @@
+* For seeding packages to this VM you can use White-listing or mirror policy.
+* Bring this VM (npmSource) to the closed network.
+* Also bring npmBlank that has no packages on it to the closed network. (It’s just a blank installation of npm Enterprise).
+* Now set up the blank npm Enterprise (i.e npmBlank) to mirror the npmSource VM with your packages on it.
```
Please change this block to:

To initialize your installation:

 1. With `npmeSource` still connected to the internet, edit the `whitelist` file so that it's a new-line delimited list of the package names you'll want available on your closed network. The `white-list` file is in the data folder which is `/usr/local/lib/npme/data` by default unless you changed this under `Miscellaneous Data` in the npme Settings Panel.
 1. Allow adequate time for the replication process to pull in all the packages added to the `whitelist` file.
 1. Bring `npmeSource` and `npmePrimary` into the closed network.
 1. In `npmePrimary`'s Configuration Settings, under `Upstream Registry`:
    1. point the Upstream URL to `npmeSource`
    1. set the `Upstream Secret` to match `npmeSource`'s
    1. set the `Policy` to `mirror`

arobson commented on tutorials/npm-Enterprise-on-a-closed-network.md:
```diff
@@ -0,0 +1,62 @@
+* Also bring npmBlank that has no packages on it to the closed network. (It’s just a blank installation of npm Enterprise).
+* Now set up the blank npm Enterprise (i.e npmBlank) to mirror the npmSource VM with your packages on it.
+
+Follow the steps below to bring new packages from public Internet to your closed network.
```
Please change this block to:

To add new packages from the public registry to your closed network:

 1. Clone `npmeSource` and allow it access to the internet.
 1. Add any new packages to the `whitelist` that were missing from the initial install.
 1. Copy `/usr/local/lib/npme` to a memory stick or portable drive. If you changed the location of the storage paths, be sure to copy the contents from _all_ the folders.
 1. Stop `npmeSource`'s npme instance that's running inside your network from the npme Dashboard.
 1. Copy over all the files from the cloned instance to the correct destination path on `npmeSource`.
 1. Restart the npme instance on `npmeSource` from the npme Dashboard.
 1. Restart replication on `npmePrimary` with the command `npme reset-follower` from the shell.

arobson commented on tutorials/npm-Enterprise-on-a-closed-network.md:
```diff
@@ -0,0 +1,62 @@
+
+* Install Docker on the VM running on a closed network.
+
+Install a supported version of Docker on your server. Replicated supports Docker from 1.7.1 to 1.13.1. We recommend installing the latest version of Docker available for the range of your [operating system](https://www.replicated.com/docs/kb/supporting-your-customers/installing-docker-in-airgapped/).
```
This is incorrect. Replicated supports up to `17.03.1-ce` (according to their site) and is likely to continue keeping up with releases. Please change "Replicated supports Docker from 1.7.1 to 1.13.1." to "At the time of this writing, Replicated still supported Docker versions as old as 1.7.1."

arobson reviewed (changes requested):
@mahikhond Thanks for the work you put into this. The edits I've requested are so extensive because of how complex the steps are in order to get the first approach to work and I'd like to be as specific as possible in order to avoid follow-up tickets. Thanks!

mahikhond committed:
fix: Updated suggested changes.

Details:
Added minor changes in the document for npm Enterprise on closed network.
